src/Esprima/Scanner.cs

```diff
@@ -1549,6 +1549,13 @@ public Token ScanTemplate()
                 tmp = Regex.Replace(tmp, "[\uD800-\uDBFF][\uDC00-\uDFFF]", astralSubstitute);
             }
 
+            // .NET doesn't support [^] which is equivalent to `[^.]`
+            // c.f. https://github.com/sebastienros/esprima-dotnet/issues/146
+            if (tmp.Contains("[^]"))
```

Comment:
not sure if calling Replace directly is much different, search must be done again

Reply:
I just checked and it looks like it's allocating in all cases. https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs#L1042

It's probably insignificant, but I think we can keep the test then.

Reply:
Sure, I was just wondering.